### PR TITLE
Update Calcite version

### PR DESCRIPTION
**Pull request by mihaibudiu**
Also updated docker images, since the new calcite has upgraded gradle, which is in the docker image.

### CODE REVIEW

**Review by gz — APPROVED** [review]

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/SqlToRelCompiler.java [hunk 1]
```diff
@@ -2128,7 +2128,7 @@ public CreateViewStatement compileCreateView(
 
         // Convert plan to use checked arithmetic
         // Must be done before optimizations.
-        ConvertToChecked checkedConverter = new ConvertToChecked(this.getRexBuilder());
+        ConvertToChecked checkedConverter = new ConvertToChecked(this.getRexBuilder(), true);
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/SqlToRelCompiler.java:2131]: Carrying forward from #6177: this `true` is a magic boolean at a public call site. What does the new second arg to `ConvertToChecked` mean? Either a `// <name>: true` argument-name comment here, or — better — a named constant / enum, so a reader can tell what it controls without chasing it into Calcite.

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/optimizer/RexOptimize.java [hunk 2]
```diff
@@ -64,18 +68,22 @@ RexNode optimizeInterval(RexCall call) {
...
                         }
+                        case CHECKED_TIMES:
                         case TIMES: {
                             BigDecimal result = leftValue.multiply(rightValue);
                             return this.builder.makeLiteral(result, resultType, true, false);
                         }
+                        case CHECKED_DIVIDE:
                         case DIVIDE: {
                             // Microsecond precision; values are in milliseconds, and we keep 3 more decimals
                             BigDecimal result = leftValue.divide(rightValue, 3, RoundingMode.FLOOR);
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/optimizer/RexOptimize.java:89]: Carrying forward from #6177: the new `CHECKED_PLUS / CHECKED_MINUS / CHECKED_TIMES / CHECKED_DIVIDE` constant-folding branches still have no test. The only new test in this PR (`calciteIssue7501`) is about LEFT JOIN/USING, not interval arithmetic. Constant folding of checked ops needs at least one test per kind that exercises a literal±literal interval, and ideally one that exercises the overflow path — does this folder still produce a literal when the result would overflow, or should it bail out and let the runtime check trigger? As written we can't tell whether folding silently bypasses the overflow check that `CHECKED_*` exists to enforce.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression2Tests.java [hunk 3]
```diff
@@ -1013,4 +1013,22 @@ public void issue5927a() {
                  d| 1""");
         ccs.visit(this.findLinear(ccs.compiler));
     }
+
+    @Test
+    public void calciteIssue7501() {
+        showPlan();
+        this.getCC("""
+                CREATE TABLE D(sk_cid INT, dt DATE, dm_sym VARCHAR, fhd DATE);
+                CREATE TABLE F(sk_cid INT);
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression2Tests.java:1021]: `showPlan()` flips a JVM-global logger level (`Logger.INSTANCE.setLoggingLevel(SqlToRelCompiler.class, 2)`) and never resets it. Every test that runs after `calciteIssue7501` in the same JUnit JVM will dump plans to stdout. Looks like leftover debug — please drop the call (or wrap it in a try/finally that restores the previous level).

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/QATests.java [hunk 4]
```diff
@@ -25,6 +25,7 @@ static void compileAndCheck(File file) throws SQLException {
 
     @Test
     public void qaTests() throws SQLException {
+        // BaseSQLTests.showPlan();
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/QATests.java:28]: Commented-out debug line — please drop it rather than commit it. Broken-window territory.

File: sql-to-dbsp-compiler/build.sh [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-set -e
+set -ex
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/build.sh:3]: `set -ex` (was `set -e`) turns on shell tracing for every CI build that calls this script — every line gets echoed to stderr. Was that intentional, or leftover from debugging the gradle/JVM detection? If the latter, please revert to `set -e`.

File: sql-to-dbsp-compiler/build.sh [hunk 6]
```diff
@@ -57,7 +57,8 @@ if [ "${CALCITE_BUILD_NEXT}" = "y" ]; then
     fi
 
     pushd "${CALCITE_BUILD_DIR}" >/dev/null
-    jvm_version=$(./gradlew --version | sed -n 's/^JVM: *//p')
+    ./gradlew --version
+    jvm_version=$(./gradlew --version | sed -n 's/^Launcher JVM: *//p')
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/build.sh]: `./gradlew --version` is now invoked twice back-to-back (once for the log, once captured for `jvm_version`). Each invocation spins up a daemon. Capture once: `out=$(./gradlew --version); echo "$out"; jvm_version=$(echo "$out" | sed -n 's/^Launcher JVM: *//p')`.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/CatalogTests.java [hunk 7]
```diff
@@ -8,7 +8,6 @@
 import org.dbsp.sqlCompiler.circuit.operator.DBSPSinkOperator;
 import org.dbsp.sqlCompiler.compiler.CompilerOptions;
 import org.dbsp.sqlCompiler.compiler.DBSPCompiler;
-import org.dbsp.sqlCompiler.compiler.backend.MerkleOuter;
 import org.dbsp.sqlCompiler.compiler.backend.rust.ToRustVisitor;
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/CatalogTests.java:11]: Unrelated cleanup (unused-import removal) bundled into a Calcite-bump PR. Trivial, but ideally a separate commit so `git log` for this file tells a clean story.

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/SqlToRelCompiler.java (hunk 1 above)
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/SqlToRelCompiler.java:2131]: Magic boolean second arg still here. Reads as a bug at the call site — please name it (`/* failOnError= */ true` or whatever the parameter actually means) or expose a constant on `ConvertToChecked`.

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/optimizer/RexOptimize.java [hunk 8]
```diff
@@ -54,6 +54,10 @@ RexNode optimizeInterval(RexCall call) {
                     return this.builder.makeLiteral(value.negate(), resultType, true, false);
                 }
             }
+            case CHECKED_PLUS:
```
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/optimizer/RexOptimize.java:57]: These new `CHECKED_*` folding branches are still untested. The only new test (`calciteIssue7501`) is a LEFT JOIN / USING reproducer — it doesn't exercise constant folding of checked interval arithmetic at all. Worth a small test that folds e.g. `INTERVAL '...' + INTERVAL '...'` under the checked plan and asserts the result, *and* one where the operation would overflow — the current branch produces a literal unconditionally via `BigDecimal`, which silently bypasses the runtime overflow check that `CHECKED_*` exists to enforce.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/QATests.java (hunk 4 above)
Comment (mythical-fred) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/QATests.java:28]: Commented-out `// BaseSQLTests.showPlan();` is debug residue — drop it. Same broken-window energy as the `showPlan()` that previously slipped into `calciteIssue7501`.

File: sql-to-dbsp-compiler/build.sh (hunk 5 above)
Comment (mythical-fred) [inline sql-to-dbsp-compiler/build.sh:3]: `set -ex` adds shell tracing to every CI invocation of this script. Looks like leftover debugging from chasing the gradle/JVM detection — drop the `x` unless you actually want xtrace in production CI logs forever.
